### DISCUSSION TITLE
How to create a new micropython type in a c module

### DISCUSSION THREAD

**Discussion by uraich** [micropython/micropython#12321] [Core Development]
Hi,
mp_obj_type_t seems to have changed recently and it uses slot_index_xxx now. Can you please point me to an example, how to create a new MP type.
I am trying to get the interface for tensorflow going and there the old way is used, which does not work any longer .
![image](https://github.com/micropython/micropython/assets/1673533/8f459b78-1da7-4141-95a3-d92feb9e56fd)

**Comment by jimmo** [comment]
https://github.com/micropython/micropython/wiki/Build-Troubleshooting#define_const_obj_type

**Comment by uraich** [comment]
Thanks, that did the trick!
